File: firebase/src/main/java/com/admin/theapp/firebase/utils/mappers/FirebaseHotelModelToHotelMapper.java
package com.admin.theapp.firebase.utils.mappers;

import android.support.annotation.NonNull;

import com.admin.theapp.Hotel;
import com.admin.theapp.firebase.model.FirebaseHotelModel;
import com.theapp.tools.Mapper;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class FirebaseHotelModelToHotelMapper extends Mapper<FirebaseHotelModel, Hotel> {

    @Inject
    FirebaseHotelModelToHotelMapper() { // for dagger2
    }

    @NonNull
    @Override
    public Hotel map(@NonNull FirebaseHotelModel firebaseHotelModel) {
        return new Hotel(
                firebaseHotelModel.getId(),
                firebaseHotelModel.getName(),
                firebaseHotelModel.getAddress(),
                firebaseHotelModel.getStars(),
                firebaseHotelModel.getDistance(),
                firebaseHotelModel.getImage(),
                firebaseHotelModel.getSuites_availability(),
                firebaseHotelModel.getLat(),
                firebaseHotelModel.getLon()
        );
    }
}

File: storage/src/main/java/com/admin/theapp/firebase/utils/mappers/FirebaseHotelModelToHotelMapper.kt
package com.admin.theapp.firebase.utils.mappers

import com.admin.theapp.Hotel
import com.admin.theapp.firebase.entities.FirebaseHotelModel
import com.theapp.tools.Mapper
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class FirebaseHotelModelToHotelMapper @Inject constructor() : Mapper<FirebaseHotelModel, Hotel>() {

    override fun map(from: FirebaseHotelModel) =
            Hotel(from.id, from.name, from.address, from.stars, from.distance,
                    from.image, from.suites_availability, from.lat, from.lon)
}
